pyk/src/pyk/kast.py

```diff
@@ -210,6 +221,9 @@ def unapply(self, term: KInner) -> KInner:
             new_term = KRewrite(lhs, rhs).replace(new_term)
         return new_term
 
+    def pretty_print(self, kprint) -> str:
```

Comment:
What do you think about `pretty` or `pretty_str`? For me, `print` suggests side effects.

Reply:
I like `pretty`.

Reply:
I've changed this to return an `Iterator[str]` of lines. This makes things like indentation much easier, and matches what we're doing in `KCFG`.

Reply:
I think the return type should remain `str`. If the string needs further processing, splitting it at newlines before doing all the other stuff should be simple, but e.g. for debugging in an interpreter joining the strings all the time is cumbersome.

Reply:
There's another functional advantage of `Iterable[str]`. When pretty printing large CFGs, if we use `str` (or `List[str]`) we need to render the entire subtree at each branch before adding indentation. This creates a noticeable lag (I've seen as much as 5 seconds). If we use `Iterable[str]` then we may use generators/yield make `pretty` lazy and allow a better user experience.